pandas/core/frame.py

```diff
@@ -8387,6 +8387,48 @@ def isin(self, values):
                 self.columns,
             )
 
+    def value_counts(self):
+        """
+        The number of times each unique row appears in the DataFrame.
+
+        Returns
+        -------
+        counts : Series
+
+        See Also
+        --------
+        Series.value_counts: Equivalent method on Series.
+
+        Examples
+        --------
+
+        >>> df = pd.DataFrame({'num_legs': [2, 4, 4], 'num_wings': [2, 0, 0]},
+        ...                   index=['falcon', 'dog', 'cat'])
+        >>> df
+                num_legs  num_wings
+        falcon         2          2
+        dog            4          0
+        cat            4          0
+
+        >>> df.value_counts()
+        (4, 0)    2
+        (2, 2)    1
+        dtype: int64
+
+        >>> df1col = df[['num_legs']]
+        >>> df1col
+                num_legs
+        falcon         2
+        dog            4
+        cat            4
+
+        >>> df1col.value_counts()
+        (4,)    2
+        (2,)    1
+        dtype: int64
+        """
+        return self.apply(tuple, 1).value_counts()
```

Comment:
Interesting approach though I don't think this will work with NA values.

```python
>>> df = pd.DataFrame({'num_legs': [2, np.nan, np.nan], 'num_wings': [2, 0, 0]}, index=['falcon', 'dog', 'cat'])
>>> df.apply(tuple, 1).value_counts()
(2.0, 2.0)    1
(nan, 0.0)    1
(nan, 0.0)    1
dtype: int64
```

Dropping rows where any value is NA *might* be what we need (would get dropped by Series)

Reply:
With the new implementation this is what happens now:

```python
>>> df = pd.DataFrame({'num_legs': [2, np.nan, np.nan], 'num_wings': [2, 0, 0]}, index=['falcon', 'dog', 'cat'])
>>> df.value_counts()
num_legs  num_wings
2.0       2            1
dtype: int64
```

I've added a note to the docs that nan-containing rows get dropped.

Do you want to support a `dropna=True` parameter, similar to the value_counts method on Series? If you do, I do not know how to implement that functionality, and perhaps it would be best left as a follow-up issue.